### PR TITLE
Import proposed to be closed schools from GIAS

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

There are a small amout of schools that are proposed to be closed in the future but are currently still open. These schools should still be able to claim funding for their mentors.

So this commit changes the GIAS importer to accept schools that are proposed to be closed.

Work will follow to restrict these schools after they have actually been closed.

This will allow support users to onboard these soon-to-be-closed schools.

## Changes proposed in this pull request

Gias importer accepts proposed to be closed schools

## Guidance to review

Review Code

## Screenshots

This school is proposed to be closed
Screenshot from 2024-06-12 11-24-46

### CODE REVIEW

**Comment by CatalinVoineag** [comment]
[Change filter on system to allow schools with status 'proposed to close' to be added to the system.](https://trello.com/c/YAJoCspw/531-change-filter-on-system-to-allow-schools-with-status-proposed-to-close-to-be-added-to-the-system)

File: app/services/gias/csv_transformer.rb [hunk 1]
```diff
@@ -56,7 +56,7 @@ def school_in_scope?(row)
     end
 
     class School
-      OPEN_SCHOOL = "1".freeze
+      OPEN_SCHOOLS = %w[1 3].freeze # Open and proposed to be closed but still open
```
Comment (Nitemaeric) [inline app/services/gias/csv_transformer.rb:59]: Nitpick, but could we name this `OPEN_SCHOOL_CODES` to be explicit?

**Review by Nitemaeric — APPROVED** [review]

**Comment by Nitemaeric** [comment]
Additionally, I think there's likely a spec file we should update?

File: app/services/gias/csv_transformer.rb [hunk 2]
```diff
@@ -56,7 +56,7 @@ def school_in_scope?(row)
     end
 
     class School
-      OPEN_SCHOOL = "1".freeze
+      OPEN_SCHOOL_CODES = %w[1 3].freeze # Open and proposed to be closed but still open
```
Comment (ollietreend) [inline app/services/gias/csv_transformer.rb:59]: Can you please annotate what the numeric codes mean? For example, see `NON_ENGLISH_ESTABLISHMENTS` below.

https://github.com/DFE-Digital/itt-mentor-services/blob/4392230bbb5de69a64eca6483fc53876223c5c0f/app/services/gias/csv_transformer.rb#L60-L64

This'll help us understand what the codes mean in future, when we've all forgotten what `1` and `3` mean 🙈

**Pull request merged**